### PR TITLE
[BACKLOG-5807] Port the bar chart to VizAPI 3.0 (minimum set of features for demo)

### PR DESCRIPTION
**Pull request by dcleao**
@pentaho/millenniumfalcon please review.

### CODE REVIEW

File: package-res/resources/web/pentaho/type/facets/DiscreteDomain.js [hunk 1]
```diff
@@ -15,10 +15,12 @@
  */
 define([
   "./Refinement",
+  "../list",
   "../../i18n!types",
```
Comment (nantunes) [inline package-res/resources/web/pentaho/type/facets/DiscreteDomain.js]: Inexistent properties file (pentaho/type/facets/i18n/types.properties).

File: package-res/resources/web/pentaho/visual/ccc/sunburst/model.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../abstract/model",
+  "pentaho/i18n!../abstract/i18n/model",
+  "../abstract/types/displayUnits",
+  "../abstract/types/labelsOption",
+  "../abstract/types/sliceOrder",
+  "../abstract/themes"
```
Comment (nantunes) [inline package-res/resources/web/pentaho/visual/ccc/sunburst/model.js]: Should it be `"../abstract/theme/model"`? It fails like this.

File: package-res/resources/web/pentaho/visual/ccc/boxplot/model.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../categoricalContinuousAbstract/model",
+  "pentaho/i18n!../abstract/i18n/model",
+  "../abstract/themes"
```
Comment (nantunes) [inline package-res/resources/web/pentaho/visual/ccc/boxplot/model.js]: Should it be `"../abstract/theme/model"`? It fails like this.

File: package-res/resources/web/pentaho/type/facets/DiscreteDomain.js (hunk 1 above)
Comment (dcleao) [inline package-res/resources/web/pentaho/type/facets/DiscreteDomain.js]: This was not being detected due to an error in the i18n plugin. Fixed the path.

File: package-res/resources/web/pentaho/visual/ccc/sunburst/model.js (hunk 2 above)
Comment (dcleao) [inline package-res/resources/web/pentaho/visual/ccc/sunburst/model.js]: Thanks! Rebase/Merge error.

**Comment by nantunes** [comment]
:+1: All ok now.

**Pull request merged**